substrate/frame/safe-mode/src/lib.rs

```diff
@@ -15,6 +15,63 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//! # Safe Mode
+//!
+//! Trigger for stopping all extrinsics outside of a specific whitelist.
+//!
+//! ## WARNING
+//!
+//! NOT YET AUDITED. DO NOT USE IN PRODUCTION.
+//!
+//! ## Pallet API
+//!
+//! See the [`pallet`] module for more information about the interfaces this pallet exposes,
+//! including its configuration trait, dispatchables, storage items, events, and errors.
+//!
+//! ## Overview
+//!
+//! Safe mode is entered via two paths (deposit or forced) until a set block number.
+//! The mode is exited when the block number is reached or a call to one of the exit extrinsics is
+//! made. A `WhitelistedCalls` configuration contains all calls that can be executed while in
+//! safe mode.
+//!
+//! ### Primary Features
+//!
+//! - Entering safe mode can be permissionless with deposit, origin based, or some combination of
+//!   those.
+//! - Separate origin configuration for entering and exiting safe mode.
+//! - Safe mode must always be configured to exit at a specific block number.
+//! - Safe mode exit block number may be extended by additional calls.
```

Comment:
Structural consistency. Also I don't see how you can have a combination of deposit and origin?

```suggestion
//! This pallet provides:
//!
//! - Ability to enter safe mode with a privileged origin or by anyone placing a deposit;
//! - Separate origin configuration items for entering and exiting safe mode;
//! - Configuration item to set a safe mode duration, after which the system will exit safe mode;
//! - Ability to extend safe mode beyond its configured exit, if needed.
```

Reply:
The combo is possible as one of the options in the following true table:

- `Force*Origin` set & Deposit = Some
- `Force*Origin` set & Deposit = None
- `Force*Origin` root/none & Deposit = Some
- (And well no reason to set both to none)

Revised the bullet points some based on your suggestions (incoming in next commit)

Reply:
Ah, well I understood it as you could partially supply the deposit and then use some less-privileged origin.

I still think what I wrote is better. Of course the privilege origin/deposit can be practically unattainable. Even in pallets that don't allow setting `Deposit = None`, you could still configure it to be greater than the total issuance. So what you are saying in your comment applies to everything. IMO it reduces clarity to try to finesse your way around that, just document the configuration items properly.

Reply:
Here's the update from the last commit that tries to clarify. Mostly trying to highlight to someone who is looking for only one or the other that either setup is possible.

```
//! - Entering safe mode can be with a privileged origin or anyone placing a deposit.
//! - Deposit method or privileged origin may be separately disabled.
//! - Separate privileged origin configuration items for entering and exiting safe mode.
//! - Configuration item to set a safe mode block count duration, after which the system will exit
//!   safe mode.
//! - Safe mode may be extended beyond the configured exit by additional calls.
```

Reply:
To me yours is hard to read because the items take many different forms. Read them out loud, the third and fourth ones jump out because they are not sentences (there is no conjugated verb). (First has a typo BTW, "can be with".) I am more of a _snoot_ than others, will stop arguing from here, but this stuff bothers me :upside_down_face: 

IMO your second point is not necessary, the "or" in the first expresses it, and each configuration item's docs should give details on specific configurations. This is "Primary Features", not details about every config option or implementation details. It's very common to have config items have a setting that practically disables them. Is there a reason to highlight this at this level? I don't think the option to disable a config item would stop someone from using the pallet.

Reply:
It bothers me as well, so please continue. I'm just not great at it :)

I think the dual use cases is important and not quickly clear from the configuration options, but I agree that the "or" does the job.

How does this sound:

```
//! - Entering safe mode can be via privileged origin or anyone who places a deposit.
//! - Origin configuration items are separated for privileged entering and exiting safe mode.
//! - A configured block count duration, after which the system will exit safe mode.
//! - Safe mode may be extended beyond the configured exit by additional calls.
```

Reply:
I would just change the third to:
```
//! - A configurable duration sets the number of blocks after which the system will exit safe mode.
```

Reply:
Updated. Thanks for the help!